crypto_sentiment_demo_app/crawler/crawler.py

```diff
@@ -62,16 +92,154 @@ def parse_bitcointicker(self) -> pd.DataFrame:
                 "source": sources,
                 "pub_time": pub_times,
             }
-        ).drop_duplicates(
-            subset="title_id"
-        )  # TODO resolve duplications better
+        )
 
-        # TODO apply filters
-        pass
-
-        df.set_index("title_id", inplace=True)
         return df
 
+    @staticmethod
+    def get_lang_detector_spacy(nlp: SpacyEnglishPipeline, name: str):
+        """
+        A function to be used to add a Spacy language detector pipeline.
+        Taken from here:
+        https://stackoverflow.com/questions/66712753/how-to-use-languagedetector-from-spacy-langdetect-package
+        :param nlp:
+        :param name: pipeline name
+        :return:
+        """
+        return LanguageDetector()
+
+    @staticmethod
+    def __enable_langdetect_with_spacy(spacy_model: SpacyEnglishPipeline):
+        Language.factory("language_detector", func=Crawler.get_lang_detector_spacy)
+        spacy_model.add_pipe("language_detector", last=True)
+
+    @staticmethod
+    def __get_rss_urls(path_to_rss_feed_list: str) -> List[str]:
+        """
+        Gets a list of URLs form a text file
+        :param path_to_rss_feed_list: path to a text file
+        :return: a list of strings
+        """
+        with open(path_to_rss_feed_list) as f:
+            urls = [line.strip() for line in f.readlines()]
+        return urls
+
+    @staticmethod
+    def __filter_out_question_marks(df: pd.DataFrame, text_col_name: str = "title") -> pd.DataFrame:
+        """
+        A primitive method returning True if a question mark is found in the text
+        :param df: a DataFrame
+        :param text_col_name: column name to filter on
+        :return: bool
+        """
+        return df.loc[~df[text_col_name].str.contains("\?")]
```

Comment:
```suggestion
        return df.loc[~df[text_col_name].str.contains(r"\?")]
```
См. коммент про flake W605 выше